addons/website_sale/controllers/main.py

```diff
@@ -1084,6 +1084,12 @@ def shop_payment_validate(self, transaction_id=None, sale_order_id=None, **post)
         """
         if sale_order_id is None:
             order = request.website.sale_get_order()
+            if not order and 'sale_last_order_id' in request.session:
+                # Retrieve the last known order from the session if the session key `sale_order_id`
+                # was prematurely cleared. This is done to prevent the user from updating their cart
```

Comment:
I think you mean "if the session key `sale_order_id` was **not** prematurely cleared

Reply:
No, `sale_last_order_id` is used when `sale_order_id` was indeed prematurely cleared.

Ignoring paid carts altogether is a bit hack-hacky but I didn't find another solution that doesn't require refactoring all the controllers :/

Reply:
indeed I confused sale_order_id and sale_last_order_id